apps/language_server/lib/language_server/build.ex

```diff
@@ -136,6 +141,7 @@ defmodule ElixirLS.LanguageServer.Build do
         # The project may override our logger config, so we reset it after loading their config
         logger_config = Application.get_all_env(:logger)
         Mix.Task.run("loadconfig")
+        # NOTE: soft-deprecated in v1.10
```

Comment:
I would change this to FIXME since it's a latent bug for versions > 1.10.

Reply:
According to Elixir's deprecation policy we have 3 minor versions until it will even be hard-deprecated so I think a note is fine for now (or perhaps it could just be removed).: https://hexdocs.pm/elixir/compatibility-and-deprecations.html#policy

It's replacement `Application.put_all_env` was added in 1.9 so in order to cleanly handle the deprecation we'll need to depend on 1.9 which we don't want to do for a while yet.